src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,113 @@
+            statement.setBigDecimal(2, book.getPrice());
+            checkAffectedRows(statement.executeUpdate());
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getLong(1);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to put the book into the db " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (PreparedStatement statement = createStatement(FIND_BY_ID_QUERY)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.getResultSet();
+            if (resultSet.next()) {
+                return Optional.of(parseBookFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to find the book by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (PreparedStatement statement = createStatement(SELECT_All_QUERY)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> books = new ArrayList<>();
+            while (resultSet.next()) {
+                books.add(parseBookFromResultSet(resultSet));
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to select all books form DB", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (PreparedStatement statement = createStatement(UPDATE_QUERY)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            checkAffectedRows(statement.executeUpdate());
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to update the book " + book, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (PreparedStatement statement = createStatement(DELETE_BY_ID_QUERY)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() >= MIN_AFFECTED_ROWS;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to delete book by id " + id, e);
+        }
+    }
+
+    private Book parseBookFromResultSet(ResultSet resultSet) throws SQLException {
+        Book newBook = new Book();
+        newBook.setId(resultSet.getLong(ID_COLUMN));
+        newBook.setTitle(resultSet.getString(TITLE_COLUMN));
+        newBook.setPrice(resultSet.getBigDecimal(PRICE_COLUMN));
+        return newBook;
+    }
+
+    private void checkAffectedRows(int rows) {
+        if (rows < MIN_AFFECTED_ROWS) {
+            throw new DataProcessingException();

```

Comment:
It's used in two different methods with their own messages in the catch blocks. But I'll change the thrown exception class to SQLException so it will be thrown correctly